fix (docs): renamed platform-related terms (#2610)
commercetools/merchant-center-application-kit: website/src/content/what-is-a-custom-application.mdx

Using Custom Applications enables merchants to:

> * Extend new or existing functionalities of the Composable Commerce APIs and the Merchant Center.

Reviewer: ```suggestion
* Extend the functionality of the Composable Commerce APIs and the Merchant Center
```

> * Tailor specific business requirements and needs of merchants.

Reviewer: ```suggestion
* Tailor to specific business requirements and use-cases
```

> * Integrate the Composable Commerce APIs and the Merchant Center with external services.

Reviewer: ```suggestion
* Integrate Composable Commerce and the Merchant Center with external services
```

Author: @melindafekete thanks for the suggestion! What bothers me is that if we do it like that, it would be the only place where we separate Composable Commerce and the MC, since in other places we say "Composable Commerce APIs and the Merchant Center". What do you think?

Reviewer: Good point, let's keep the original change 👍
